src/SlackRTMHandler.ts

```diff
@@ -211,13 +211,26 @@ export class SlackRTMHandler extends SlackEventHandler {
 
         const isIm = chanInfo.channel.is_im || chanInfo.channel.is_mpim;
 
+        if (chanInfo.channel.is_im) {
+            // Check if the user is denied slack DMs
+            if (this.main.config.puppeting?.dms_deny?.slack) {
+                const banned = this.main.config.puppeting.dms_deny.slack.find((r) => r.test(chanInfo.channel.user!));
+                if (banned) {
+                    log.debug(`Slack user '${chanInfo.channel.user}' is disallowed from DMing, not creating room.`);
+                }
+                return;
```

Comment:
We can't. It's a DM so we can't really say anything to them without stealing the users account. Since it's a puppeted account, they can still respond via Slack.